lib/rubocop/cop/style/class_structure.rb

```diff
@@ -0,0 +1,312 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks if the code style follows the ExpectedOrder configuration:
+      #
+      # `Categories` allows us to map macro names into a category.
+      #
+      # Consider an example of code style that covers the following order:
+      # - Constants
+      # - Associations (has_one, has_many)
+      # - Attributes (attr_accessor, attr_writer, attr_reader)
+      # - Initialize
+      # - Instance methods
+      # - Protected methods
+      # - Private methods
+      #
+      # You can configure the following order:
+      #
+      # ```yaml
+      #  Style/ClassStructure:
+      #    ExpectedOrder:
+      #      - constant
+      #      - association
+      #      - attribute
+      #      - initialize
+      #      - instance_method
+      #      - protected_method
+      #      - private_method
+      #
+      # ```
+      # Instead of putting all literals in the expected order, is also
+      # possible to group categories of macros.
+      #
+      # ```yaml
+      #  Style/ClassStructure:
+      #    Categories:
+      #      association:
+      #        - has_many
+      #        - has_one
+      #      attribute:
+      #        - attr_accessor
+      #        - attr_reader
+      #        - attr_writer
+      # ```
+      #
+      # @example bad code
```

Comment:
I'm not sure if I think this cop should include examples. They would be so long and configuration dependent that it becomes a bit unclear what is good and bad about them. 🤔

Reply:
I think this is bad for the current setup configuration. Maybe, we can add more examples of setups to show how it works?

Reply:
Also the "bad" here I think means do not follow the default style guide. Not sure if it's clear and obvious. The end part is linking the official documentation and I steal this class example from there.